jfxtras-agenda/src/main/java/jfxtras/internal/scene/control/skin/agenda/base24hour/AgendaSkinTimeScale24HourAbstract.java

```diff
@@ -238,7 +241,15 @@ public void refresh() {
 		refreshLocale();
 		setupAppointments();
 		nowUpdateRunnable.run(); 
-	}
+	}
+
+	/**
+	 * 
+	 */
+	@Override
+    public Pane getNodeForPopup(Appointment appointment) { return appointmentNodeMap.get(appointment); }
+    final private Map<Appointment, Pane> appointmentNodeMap = new WeakHashMap<>();
```

Comment:
Encapsulation would suggest a separate listener, since it would be part of a different class.